### PR TITLE
[JBIDE-25108][JBIDE-25126] validate server editor and save only if valid

### PR DESCRIPTION
**Pull request by adietish**
* corrected: validators for devmode- and debug-key 
(how to replicate: provide a key with special characters, save the editor and restart the adapter. It will complain that the key(s) cannot be set and that they have to comply to ```
start with alphabetic character or '_', followed by a string of alphanumeric characters or '_' (e.g. 'my_name',  or 'MY_NAME',  or 'MyName', regex used for validation is '[A-Za-z_][A-Za-z0-9_]*')``` 
* corrected: form shows error status in header

this depends on https://github.com/jbosstools/jbosstools-base/pull/601 to be merged into jbosstools-base beforehand.

# Pull Request Checklist
## General
* Is this a blocking issue or new feature? If yes, QE needs to +1 this PR

## Code
* Are method-/class-/variable-names meaningful?
* Are methods concise, not too long?
* Are catch blocks catching precise Exceptions only (no catch all)?

## Testing
* Are there unit-tests?
* Are there integration tests (or at least a jira to tackle these)?
* Is the non-happy path working, too?
* Are other parts that use the same component still working fine?

## Function
* Does it work?

### CODE REVIEW

**Comment by bdshadow** [comment]
testPR

**Comment by adietish** [comment]
@jeffmaury thanks for spotting this. I can replicate it. Fixing.

**Comment by adietish** [comment]
@jeffmaury fixed, please test the latest commit

File: plugins/org.jboss.tools.openshift.common.ui/src/org/jboss/tools/openshift/internal/common/ui/databinding/DisablableMultiValidator.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import org.eclipse.core.runtime.IStatus;
+
+/**
+ * A multi validator that validates a given value observable if validation is
+ * enabled by the given enablement observable. Validation happens based on the
+ * given validation function.
+ * 
+ * @author Andre Dietisheim
+ */
+public class DisablableMultiValidator<T> extends MultiValidator {
```
Comment (jeffmaury) [inline plugins/org.jboss.tools.openshift.common.ui/src/org/jboss/tools/openshift/internal/common/ui/databinding/DisablableMultiValidator.java]: Typo in the className ?

File: plugins/org.jboss.tools.openshift.common.ui/src/org/jboss/tools/openshift/internal/common/ui/utils/FormUtils.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+ * Contributors:
+ *     Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package org.jboss.tools.openshift.internal.common.ui.utils;
+
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.swt.widgets.Control;
+import org.eclipse.ui.forms.IMessage;
+
+public class FormUtils {
```
Comment (jeffmaury) [inline plugins/org.jboss.tools.openshift.common.ui/src/org/jboss/tools/openshift/internal/common/ui/utils/FormUtils.java]: Should be in base, no ?

Comment (adietish) [reply plugins/org.jboss.tools.openshift.common.ui/src/org/jboss/tools/openshift/internal/common/ui/utils/FormUtils.java]: @jeffmaury I agree, if @robstryker is willing to have it moved?

File: plugins/org.jboss.tools.openshift.common.ui/src/org/jboss/tools/openshift/internal/common/ui/databinding/DisablableMultiValidator.java (hunk 1 above)
Comment (adietish) [reply plugins/org.jboss.tools.openshift.common.ui/src/org/jboss/tools/openshift/internal/common/ui/databinding/DisablableMultiValidator.java]: @jeffmaury renaming to DisableableMultiValitdator

Comment (adietish) [reply plugins/org.jboss.tools.openshift.common.ui/src/org/jboss/tools/openshift/internal/common/ui/databinding/DisablableMultiValidator.java]: @jeffmaury btw. there are quite a few thing in openshift.commons that we could move to base. But that's for later.

**Comment by adietish** [comment]
@jeffmaury renamed DisablableMultiValidator->DisableableMultiValidator and moved FormUtils to common.ui in https://github.com/jbosstools/jbosstools-base/pull/601
Please ack/refute.

**Comment by adietish** [comment]
@jeffmaury now that https://github.com/jbosstools/jbosstools-base/pull/601 is merged and your other change request is fullfilled, +1?

**Comment by adietish** [comment]
testPR

**Comment by adietish** [comment]
jenkins is back into troubles with nexus. Build works fine locally:
```
[INFO] jbosstools-openshift ............................... SUCCESS [  0.450 s]
[INFO] openshift.plugins .................................. SUCCESS [  0.008 s]
[INFO] org.jboss.tools.openshift.egit.core ................ SUCCESS [ 30.580 s]
[INFO] org.jboss.tools.openshift.egit.ui .................. SUCCESS [  3.256 s]
[INFO] org.jboss.tools.openshift.express.client ........... SUCCESS [ 10.654 s]
[INFO] org.jboss.tools.openshift.common.core .............. SUCCESS [  0.845 s]
[INFO] org.jboss.tools.openshift.express.core ............. SUCCESS [  5.205 s]
[INFO] org.jboss.tools.openshift.common.ui ................ SUCCESS [  1.168 s]
[INFO] org.jboss.tools.openshift.express.ui ............... SUCCESS [ 11.653 s]
[INFO] org.jboss.tools.openshift.client ................... SUCCESS [  0.313 s]
[INFO] org.jboss.tools.openshift.core ..................... SUCCESS [  0.894 s]
[INFO] org.jboss.tools.openshift.ui ....................... SUCCESS [  2.022 s]
[INFO] org.jboss.tools.openshift.cdk.server ............... SUCCESS [  0.883 s]
[INFO] org.jboss.tools.openshift.js ....................... SUCCESS [  3.662 s]
[INFO] org.jboss.tools.openshift.io.core .................. SUCCESS [  4.690 s]
[INFO] org.jboss.tools.openshift.io.ui .................... SUCCESS [  3.623 s]
[INFO] openshift.itests ................................... SUCCESS [  0.006 s]
[INFO] openshift.test-framework ........................... SUCCESS [  0.009 s]
[INFO] org.jboss.tools.openshift.reddeer .................. SUCCESS [  0.864 s]
[INFO] org.jboss.tools.openshift.ui.bot.test .............. SUCCESS [  0.869 s]
[INFO] openshift.tests .................................... SUCCESS [  0.005 s]
[INFO] org.jboss.tools.openshift.egit.test ................ SUCCESS [ 18.836 s]
[INFO] org.jboss.tools.openshift.express.test ............. SUCCESS [ 17.959 s]
[INFO] org.jboss.tools.openshift.test ..................... SUCCESS [ 43.629 s]
[INFO] org.jboss.tools.openshift.cdk.server.test .......... SUCCESS [ 19.440 s]
[INFO] org.jboss.tools.openshift.io.test .................. SUCCESS [ 13.754 s]
[INFO] org.jboss.tools.cdk.reddeer ........................ SUCCESS [  0.325 s]
[INFO] openshift.features ................................. SUCCESS [  0.014 s]
[INFO] org.jboss.tools.openshift.feature .................. SUCCESS [  0.368 s]
[INFO] org.jboss.tools.openshift.test.feature ............. SUCCESS [  0.310 s]
[INFO] org.jboss.tools.openshift.egit.integration.feature . SUCCESS [  3.157 s]
[INFO] org.jboss.tools.openshift.egit.integration.test.feature SUCCESS [  0.150 s]
[INFO] org.jboss.tools.openshift.express.feature .......... SUCCESS [  0.212 s]
[INFO] org.jboss.tools.openshift.express.test.feature ..... SUCCESS [  0.192 s]
[INFO] org.jboss.tools.openshift.cdk.feature .............. SUCCESS [  0.238 s]
[INFO] org.jboss.tools.openshift.cdk.test.feature ......... SUCCESS [  0.241 s]
[INFO] org.jboss.tools.openshift.integration.tests.feature  SUCCESS [  0.227 s]
[INFO] org.jboss.tools.openshift.js.feature ............... SUCCESS [  2.673 s]
[INFO] org.jboss.tools.openshift.io.feature ............... SUCCESS [  2.556 s]
[INFO] org.jboss.tools.openshift.io.test.feature .......... SUCCESS [  0.169 s]
[INFO] openshift.site ..................................... SUCCESS [ 18.021 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 05:38 min
[INFO] Finished at: 2017-10-17T20:46:48+02:00
[INFO] Final Memory: 212M/1608M
[INFO] ------------------------------------------------------------------------
```

**Pull request merged**